sdk/src/main/java/com/hedera/hashgraph/sdk/TransactionId.java

```diff
@@ -97,8 +98,19 @@ public static TransactionId withValidStart(AccountId accountId, Instant validSta
      * @return {@link com.hedera.hashgraph.sdk.TransactionId}
      */
     public static TransactionId generate(AccountId accountId) {
-        Instant instant = Clock.systemUTC().instant().minusNanos((long) (Math.random() * 5000000000L + 8000000000L));
-        return new TransactionId(accountId, instant);
+        long currentTime;
+        long lastTime;
+
+        do {
+            currentTime = System.currentTimeMillis() * 1_000_000L;
+            lastTime = monotonicTime.get();
+            if (currentTime <= lastTime) currentTime = lastTime + 1000L;
+        }
+
+        while (!monotonicTime.compareAndSet(lastTime, currentTime));
```

Comment:
Can we add a comment here and maybe extract the consts for more code clarity

Reply:
@rokn please check it now :)